src/core/unit-varlink.c

```diff
@@ -311,18 +314,114 @@ static int list_unit_one(sd_varlink *link, Unit *unit, bool more) {
         return sd_varlink_reply(link, v);
 }
 
+static int lookup_unit_by_name(sd_varlink *link, Manager *manager, const char *name, Unit **ret_unit) {
+        assert(link);
+        assert(manager);
+        assert(ret_unit);
+
+        if (!name)
+                return -EINVAL;
+        if (!unit_name_is_valid(name, UNIT_NAME_PLAIN|UNIT_NAME_INSTANCE))
+                return -EINVAL;
```

Comment:
We should add `json_dispatch_unit_name()` or so and do this check there already

Reply:
I thought about it but stuck on what should be the mapping between UnitNameFlags and sd_json_dispatch_flags_t.

Any ideas?

Reply:
I would just make a helper specifically for the case of `UNIT_NAME_PLAIN|UNIT_NAME_INSTANCE`, maybe `json_dispatch_unit_name_plain_or_instance()`?